### PR TITLE
Add Shadcn

### PR DESCRIPTION
**Pull request by Dosbodoke**
### What is the purpose of this PR?
Add [shadcn](https://www.shadcn-vue.com/docs/installation/vite.html) as the project component library.

### What was done to achieve it?
Followed the documentation of [installing shadcn in a Vue + Vite application](https://www.shadcn-vue.com/docs/installation/vite.html)

**We will need to use the CLI in order to use this library**, this was a thing that I have pointed as as benefic to the Shadcn approach, the ability to just copy and paste the components, but now I found that the vue library only allow to install via CLI

VUE Shadcn             |  React Shadcn
:-------------------------:|:-------------------------:
VUE Shadcn |  React Shadcn

As you can see, the React library has a guide to install the components manually, and in most cases is the same steps
- Download the Radix primitive component
- Create a file with some code that use this component on the Shadcn way

I don't see this as a problem, but we will need to have a new `components.json` file at the root of the project that is needed to use the CLI. 

**About theming**, there has a [documentation](https://www.shadcn-vue.com/docs/theming.html#adding-new-colors) to do it, we can iterate more on this.

### How to test if it works?
- Go to any component and put `bg-destructive` and see if it changes the BG.
- Install any component with the CLI and try to render it. ( installed components will be placed on the `/components/ui` folder

### This change can impact another functionality?
It can change the current style, but I think this is not a concern at the moment.

### CODE REVIEW

**Comment by WaldemarLehner** [comment]
Please merge `rewrite` into this branch to fix the Pipeline.

File: tailwind.config.js [hunk 1]
```diff
@@ -1,26 +1,80 @@
...
+    './src/**/*.{ts,tsx,vue}',
+	],
+  
   theme: {
-    extend: {
+    container: {
+      center: true,
+      padding: "2rem",
       screens: {
-        'xs': '480px',
```
Comment (JulianDietzel) [inline tailwind.config.js:10]: Why was this removed?

**Comment by JulianDietzel** [comment]
Apart from the question why the "xs" breakpoint was removed, this looks good to me :+1: 

We should then address adjusting the theming according to #309 in a follow up PR.

File: tailwind.config.js (hunk 1 above)
Comment (Dosbodoke) [reply tailwind.config.js:10]: The Shadcn CLI have changed the `tailwind.config.js` automatically. But do you think it's really necessary to add one more breakpoint for smaller devices? Shouldn't we follow [the default](https://tailwindcss.com/docs/screens#configuring-custom-screens) of tailwind?

Comment (JulianDietzel) [reply tailwind.config.js:10]: @bastislack added it in #308 and I suppose he needs it. But then also he can add it back when / if he needs it. So I suppose it is fine for now.
